packages/compiler-core/src/transform.ts

```diff
@@ -180,9 +182,20 @@ export function createTransformContext(
 
     // methods
     helper(name) {
-      context.helpers.add(name)
+      const count = context.helpers.get(name) || 0
+      context.helpers.set(name, count + 1)
       return name
     },
+    removeHelper(name) {
+      const count = context.helpers.get(name)
+      if (count) {
+        const currentCount = count - 1
+        context.helpers.set(name, currentCount)
+        if (!currentCount) {
+          context.helpers.delete(name)
+        }
```

Comment:
nit: the `set` is not necessary if we are going to `delete` just after right?
Could be written as:
```
if (!currentCount) {
  context.helpers.delete(name)
} else {
  context.helpers.set(name, currentCount)
}
```